gst/gst_debug.go

```diff
@@ -164,3 +188,29 @@ func (d *DebugCategory) LogTrace(message string, obj ...*Object) {
 func (d *DebugCategory) LogMemDump(message string, obj ...*Object) {
 	d.logDepth(LevelMemDump, message, 2, getLogObj(obj...))
 }
+
+type LogFunction func(
+	level DebugLevel,
+	file string,
+	function string,
+	line int,
+	object *glib.Object,
+	message string,
+)
```

Comment:
do we need to declare a new type or would an alias suffice?

I think an alias is more comfortable to to use.

```suggestion
type LogFunction = func(
	level DebugLevel,
	file string,
	function string,
	line int,
	object *glib.Object,
	message string,
)
```

Reply:
Should be fine. I've never used aliases in go, what's the benefit?

Reply:
when you need to pass the param type, any function with the same signature is valid.

When you declare a new type, you need to cast the function with `LogFunction(myFunc)` to get the correct type.

An alternative to this would be to use an interface which allows the user even more freedom.